pkg/gather/clusterconfig/clusterconfig_test.go

```diff
@@ -537,6 +541,42 @@ func TestGatherHostSubnet(t *testing.T) {
 	}
 }
 
+func TestGatherMachineSet(t *testing.T) {
+	var machineSetYAML = `
+apiVersion: machine.openshift.io/v1beta1
+kind: MachineSet
+metadata:
+    name: test-worker
+`
+	gvr := schema.GroupVersionResource{Group: "machine.openshift.io", Version: "v1beta1", Resource: "machinesets"}
+	client := dynamicfake.NewSimpleDynamicClient(runtime.NewScheme())
+	decUnstructured := yaml.NewDecodingSerializer(unstructured.UnstructuredJSONScheme)
+
+	testMachineSet := &unstructured.Unstructured{}
+
+	_, _, err := decUnstructured.Decode([]byte(machineSetYAML), nil, testMachineSet)
+	if err != nil {
+		t.Fatal("unable to decode machineset ", err)
+	}
+	_, err = client.Resource(gvr).Create(testMachineSet, metav1.CreateOptions{})
+	if err != nil {
+		t.Fatal("unable to create fake machineset ", err)
+	}
+
+	gatherer := &Gatherer{dynamicClient: client}
+	records, errs := GatherMachineSet(gatherer)()
+	if len(errs) > 0 {
+		t.Errorf("unexpected errors: %#v", errs)
+		return
+	}
+	if len(records) != 1 {
+		t.Fatalf("unexpected number or records %d", len(records))
+	}
+	if records[0].Name != "machinesets/test-worker" {
+		t.Fatalf("unexpected machineset name %s", records[0].Name)
+	}
```

Comment:
This is pretty simple test, but I was struggling with unmarshalling the record to something meaningful (wasn't successful with `Unstructured`).